src/prognostic_equations/edmfx_closures.jl

```diff
@@ -206,78 +228,187 @@ function mixing_length(
+                l_TKE = (-(ᶜtke_exch) - sqrt(discriminant)) / (2 * a_pd)
+            end
+            l_TKE = max(l_TKE, FT(0)) # Ensure it's non-negative
+        end
+    elseif abs(ᶜtke_exch) > eps_FT # If a_pd is zero, balance is between exchange and dissipation
+        # ᶜtke_exch = c_neg / l_TKE  => l_TKE = c_neg / ᶜtke_exch
+        # Ensure division is safe and result is positive
+        if ᶜtke_exch > eps_FT # Assuming positive exchange means TKE sink from env perspective
+            l_TKE = c_neg / ᶜtke_exch # if c_neg is positive, l_TKE is positive
+        elseif ᶜtke_exch < -eps_FT # Negative exchange means TKE source for env
+            # -|ᶜtke_exch| = c_neg / l_TKE. If c_neg > 0, this implies l_TKE < 0, which is unphysical.
+            # This case (a_pd=0, tke_exch < 0, c_neg > 0) implies TKE source and dissipation, no production.
+            # Dissipation = Source. So, c_d * k_sqrt_k / l = -tke_exch. l = c_d * k_sqrt_k / (-tke_exch)
+            l_TKE = c_neg / (-(ᶜtke_exch))
+        end
+        l_TKE = max(l_TKE, FT(0))
     end
-
-    # compute l_N - the effective static stability length scale.
-    N_eff = sqrt(max(ᶜlinear_buoygrad, 0))
-    if N_eff > 0.0
-        l_N = min(sqrt(max(c_b * ᶜtke, 0)) / N_eff, l_z)
-    else
-        l_N = l_z
+    # If a_pd = 0 and ᶜtke_exch = 0 (or c_neg = 0), l_TKE remains zero.
+
+    # --- l_N: Static-stability length scale (buoyancy limit), constrained by l_z ---
+    N_eff_sq = max(ᶜlinear_buoygrad, FT(0)) # Use N^2 only if stable (N^2 > 0)
+    l_N = l_z # Default to wall distance if not stably stratified or TKE is zero
+    if N_eff_sq > eps_FT && tke_pos > eps_FT
+        N_eff = sqrt(N_eff_sq)
+        # l_N ~ sqrt(c_b * TKE) / N_eff
+        l_N_physical = sqrt(c_b * tke_pos) / N_eff
+        # Limit by distance from wall
+        l_N = min(l_N_physical, l_z)
     end
+    l_N = max(l_N, FT(0)) # Ensure non-negative
 
-    # compute l_smag - smagorinsky length scale
-    l_smag = smagorinsky_lilly_length(
-        CAP.c_smag(params),
-        N_eff,
-        ᶜdz,
-        ᶜPr,
-        ᶜstrain_rate_norm,
-    )
 
-    # add limiters
-    l = SA.SVector(
-        l_N > l_z ? l_z : l_N,
-        l_TKE > l_z ? l_z : l_TKE,
-        l_W > l_z ? l_z : l_W,
-    )
-    # get soft minimum
-    l_smin = lamb_smooth_minimum(l, smin_ub, smin_rm)
-    l_limited = max(l_smag, min(l_smin, l_z))
+    # --- Combine Scales ---
+
+    # Vector of *physical* scales (wall, TKE, stability)
+    # These scales (l_W, l_TKE, l_N) are already ensured to be non-negative.
+    # l_N is already limited by l_z. l_W and l_TKE are not necessarily.
+    l_physical_scales = SA.SVector(l_W, l_TKE, l_N)
+
+    l_smin =
+        blend_scales(scale_blending_method, l_physical_scales, turbconv_params)
+
+    # 1. Limit the combined physical scale by the distance from the wall.
+    #    This step mitigates excessive values of l_W or l_TKE.
+    l_limited_phys_wall = min(l_smin, l_z)
```

Comment:
```suggestion
```
I do not think we should have this limit after all. It undoes the stability correction on l_w in the stable case. Let's remove this unless there's a strong reason to have it.